ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -165,6 +170,82 @@ - (NSDictionary *)constantsToExport
     [view setCompassVisibility:value ? true : false];
 }
 
+RCT_EXPORT_METHOD(addPackForRegion:(nonnull NSNumber *)reactTag
+                  options:(NSDictionary*)options)
+{
+
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            if ([[options valueForKey:@"type"] isEqualToString:@"bbox"]) {
+                NSArray *b = [options valueForKey:@"bounds"];
+                MGLCoordinateBounds bounds = MGLCoordinateBoundsMake(CLLocationCoordinate2DMake([b[0] floatValue], [b[1] floatValue]), CLLocationCoordinate2DMake([b[2] floatValue], [b[3] floatValue]));
+                [mapView createOfflinePack:bounds style:[NSURL URLWithString:[options valueForKey:@"style"]] fromZoomLevel:[[options valueForKey:@"minZoomLevel"] floatValue] toZoomLevel:[[options valueForKey:@"maxZoomLevel"] floatValue] name:[options valueForKey:@"name"] type:[options valueForKey:@"type"]];
+            } else {
+                RCTLogError(@"Offline type %@ not supported. Only type `bbox` supported.", [options valueForKey:@"type"]);
+            }
+        }
+    }];
+}
+
+RCT_EXPORT_METHOD(getPacksWithCompletionHandler:(nonnull NSNumber *)reactTag
+                  findEvents:(RCTResponseSenderBlock)callback)
+{
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            RCTMapboxGL *mapView = viewRegistry[reactTag];
+            NSMutableArray* callbackArray = [NSMutableArray new];
```

Comment:
You could significantly simplify this function by taking all this code out of the block and returning this array instead of calling a completion handler.